lib/internal/stream_base_commons.js

```diff
@@ -85,7 +85,10 @@ function onWriteComplete(status) {
 
   if (status < 0) {
     const ex = errnoException(status, 'write', this.error);
-    stream.destroy(ex, this.callback);
+    if (typeof this.callback === 'function')
+      this.callback(ex);
+    else
+      stream.destroy(ex);
```

Comment:
Not sure if callback ever is anything but a function, keeping the destroy flow just in case

Reply:
I would prefer if this would be investigated, and in case we should add a test.

Reply:
I believe this is covered by `afterWriteDispatched`.